kmir/src/kmir/kdist/mir-semantics/symbolic/spl-token.md

```diff
@@ -0,0 +1,460 @@
+  rule #isSPLBorrowMutFunc("std::cell::RefCell::<&mut [u8]>::borrow_mut") => true
+  rule #isSPLBorrowMutFunc(_) => false [owise]
+
+  rule #isSPLVecSliceFunc("std::vec::Vec::<u8>::as_slice") => true
+  rule #isSPLVecSliceFunc("std::vec::Vec::<u8>::as_mut_slice") => true
+  rule #isSPLVecSliceFunc("<std::vec::Vec<u8> as std::ops::Deref>::deref") => true
+  rule #isSPLVecSliceFunc("<std::vec::Vec<u8> as std::ops::DerefMut>::deref_mut") => true
+  rule #isSPLVecSliceFunc("<std::vec::Vec<u8> as std::ops::IndexMut<std::ops::RangeFull>>::index_mut") => true
+  rule #isSPLVecSliceFunc(_) => false [owise]
+
+
+  rule #isSPLRefDerefFunc("<std::cell::Ref<'_, std::vec::Vec<u8>> as std::ops::Deref>::deref") => true
+  rule #isSPLRefDerefFunc("<std::cell::RefMut<'_, std::vec::Vec<u8>> as std::ops::Deref>::deref") => true
+  rule #isSPLRefDerefFunc("<std::cell::RefMut<'_, std::vec::Vec<u8>> as std::ops::DerefMut>::deref_mut") => true
+  rule #isSPLRefDerefFunc("<std::cell::Ref<'_, &mut [u8]> as std::ops::Deref>::deref") => true
+  rule #isSPLRefDerefFunc("<std::cell::RefMut<'_, &mut [u8]> as std::ops::Deref>::deref") => true
+  rule #isSPLRefDerefFunc("<std::cell::RefMut<'_, &mut [u8]> as std::ops::DerefMut>::deref_mut") => true
+  rule #isSPLRefDerefFunc("<core::cell::RefMut<'_, alloc::vec::Vec<u8>> as core::ops::DerefMut>::deref_mut") => true
+  rule #isSPLRefDerefFunc("<std::cell::RefMut<'_, alloc::vec::Vec<u8>> as std::ops::DerefMut>::deref_mut") => true
+  rule #isSPLRefDerefFunc("<core::cell::RefMut<'_, std::vec::Vec<u8>> as core::ops::DerefMut>::deref_mut") => true
+  rule #isSPLRefDerefFunc(_) => false [owise]
+
+  rule #isSPLAccountUnpackFunc("spl_token::state::Account::unpack_unchecked") => true
+  rule #isSPLAccountUnpackFunc("spl_token::state::Account::unpack") => true
+  rule #isSPLAccountUnpackFunc("Account::unpack_unchecked") => true
+  rule #isSPLAccountUnpackFunc(_) => false [owise]
+
+  rule #isSPLAccountPackFunc("spl_token::state::Account::pack") => true
+  rule #isSPLAccountPackFunc("Account::pack") => true
+  rule #isSPLAccountPackFunc(_) => false [owise]
+```
+
+
+## Cheatcode handling
+```{.k .symbolic}
+  rule [cheatcode-is-spl-account]:
+    <k> #execTerminator(terminator(terminatorKindCall(FUNC, operandCopy(place(LOCAL, PROJS)) .Operands, _DEST, someBasicBlockIdx(TARGET), _UNWIND), _SPAN))
+      => #setLocalValue(
+            place(LOCAL, appendP(PROJS, projectionElemDeref .ProjectionElems)),
+            Aggregate(variantIdx(0),
+              ListItem(Range(?SplAccountKey:List))                         // pub key: &'a Pubkey
+              ListItem(
+                  SPLRefCell(
+                    place(
+                      LOCAL,
+                      appendP(PROJS, projectionElemDeref projectionElemField(fieldIdx(1), #hack()) .ProjectionElems)
+                    ),
+                    Integer(?SplLamports:Int, 64, false)
+                  )
```

Comment:
What is the first parameter of `SPLRefCell` modeling? 
This value here looks like a self-reference: `#setLocalValue` above writes to `place(LOCAL, PROJS + Deref)` the value which contains a `place(LOCAL, PROJS + Deref + Field(1))`.